### PR TITLE
[WIP] Bring default dependencies under pipenv system.

### PR DESCRIPTION
**Pull request by jmchilton**
- Update sql migrate to 0.11.0.
- Update pinned and hashed variants.
- Mark Python 2 only dependencies as such, start to fill in Python 3 variants.
- Specify pipenv should use a Python 2 virtualenv to lock dependencies to make Pipfile.lock a bit less random.
- Add failing Python 3 unit test tox environment we can hack on from here (looks like nosehtml doesn't quite work yet).

Ping @nsoranzo - any interest in bringing some bits of this out of WIP?

### CODE REVIEW

File: lib/galaxy/dependencies/pipfiles/update.sh [hunk 1]
```diff
@@ -32,13 +32,14 @@ done
...
+flake8_imports
+default"
 
 for env in $ENVS
 do
         cd "$THIS_DIRECTORY/$env"
-        pipenv lock
-        pipenv lock -r > pinned-hashed-requirements.txt
+        pipenv lock --two
+        pipenv  --bare lock --two -r | sed 's/^Using.*$//' > pinned-hashed-requirements.txt
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/pipfiles/update.sh:42]: Is the `sed` for deleting the first line? It's actually leaving a first empty line, I think that `sed '1d'` should work better.

File: scripts/common_startup.sh [hunk 2]
```diff
@@ -135,17 +136,18 @@ fi
...
 fi
 
-if [ $FETCH_WHEELS -eq 1 ]; then
-    pip install -r requirements.txt --index-url "${GALAXY_WHEELS_INDEX_URL}" --extra-index-url "${PYPI_INDEX_URL}"
-    GALAXY_CONDITIONAL_DEPENDENCIES=$(PYTHONPATH=lib python -c "import galaxy.dependencies; print '\n'.join(galaxy.dependencies.optional('$GALAXY_CONFIG_FILE'))")
-    [ -z "$GALAXY_CONDITIONAL_DEPENDENCIES" ] || echo "$GALAXY_CONDITIONAL_DEPENDENCIES" | pip install -r /dev/stdin --index-url "${GALAXY_WHEELS_INDEX_URL}" --extra-index-url "${PYPI_INDEX_URL}"
+requirement_args="-r requirements.txt"
+if [ $DEV_WHEELS -eq 1 ]; then
+    requirement_args="$requirement_args -r ${GALAXY_DEV_REQUIREMENTS}"
 fi
```
Comment (nsoranzo) [inline scripts/common_startup.sh:147]: You can move lines 144-147 inside the following `if`

File: tox.ini [hunk 3]
```diff
@@ -32,6 +32,18 @@ deps =
...
+commands = bash run_tests.sh --no-create-venv -u
+whitelist_externals = bash
+setenv =
+    GALAXY_VIRTUAL_ENV = .venv3
+    GALAXY_PYTHON_VERSION=3
+deps = 
+    nose
+    NoseHTML
+    mock
+    mock-ssh-server
```
Comment (nsoranzo) [inline tox.ini:45]: Pipfile also for these?

Comment (jmchilton) [reply tox.ini:45]: Yeah - we should just throw those all in dev-dependences I think. I understand historically why I did it this way but it doesn't make sense anymore.

File: lib/galaxy/dependencies/pipfiles/default/Pipfile [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+Mako = "==1.0.2"
+pytz = "==2017.3"
+Babel = "==2.5.1"
+Beaker = "==1.7.0"
+dictobj = "==0.3.1"
+nose = "==1.3.7"
+Parsley = "==1.3"
+six = "==1.11.0"
+Whoosh = "==2.7.4"
+galaxy_sequence_utils = "==1.0.2"
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/pipfiles/default/Pipfile:50]: 1.1.2 ?

File: lib/galaxy/dependencies/pipfiles/default/Pipfile [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+# Cheetah and dependencies
+Cheetah = { version = "==2.4.4", markers="python_version < '3'" }
+Cheetah3 = { version = "==3.0.0", markers="python_version >= '3'" }
+
+# Requests and dependencies
+requests = "==2.18.4"
+certifi = "==2017.7.27.1"
+urllib3 = "==1.22"
+chardet = "==3.0.4"
+idna = "==2.6"
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/pipfiles/default/Pipfile:65]: Remove `requests` dependencies?

And similarly for the other below?

File: lib/galaxy/dependencies/pipfiles/flake8/Pipfile.lock [hunk 6]
```diff
@@ -27,23 +27,6 @@
...
-        "enum34": {
-            "hashes": [
-                "sha256:6bd0f6ad48ec2aa117d3d141940d484deccda84d4fcd884f5c3d93c23ecd8c79",
-                "sha256:644837f692e5f550741432dd3f223bbb9852018674981b1664e5dc339387588a",
-                "sha256:8ad8c4783bf61ded74527bffb48ed9b54166685e4230386a9ed9b1279e2df5b1",
-                "sha256:2d81cbbe0e73112bdfe6ef8576f2238f2ba27dd0d55752a776c41d38b7da2850"
-            ],
-            "markers": "python_version < '3.4'",
-            "version": "==1.1.6"
-        },
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/pipfiles/flake8/Pipfile.lock:46]: These changes broke the `py27-lint` tests on Travis.

**Comment by nsoranzo** [comment]
What about adding a check in `lib/galaxy/dependencies/pipfiles/update.sh` to abort if run inside the Galaxy virtualenv? I forgot to deactivate and `pipenv lock` deleted it again.

File: lib/galaxy/dependencies/pipfiles/flake8/Pipfile.lock (hunk 6 above)
Comment (jmchilton) [reply lib/galaxy/dependencies/pipfiles/flake8/Pipfile.lock:46]: I don't even know what happened to change this :(. I'm maybe pushing on this too quickly...

**Comment by jmchilton** [comment]
I'll try to find some time to break this into pieces and incorporate @nsoranzo 's comments unless someone beats me to it.

**Pull request closed without merging**